src/transformers/models/ctrl/modeling_tf_ctrl.py

```diff
@@ -254,10 +254,11 @@ def _prune_heads(self, heads_to_prune):
         """
         raise NotImplementedError
 
+    @unpack_inputs
     def call(
         self,
         input_ids=None,
-        past=None,
+        past_key_values=None,
```

Comment:
This change is well-intentioned and technically correct, but I'm going to ask you to revert `past_key_values` to `past`. It changes the public interface of the model, which may disrupt downstream users :)

Reply:
I see and understand, only problem is, without `past_key_values` in the parameters, the tests will fail. Can they both (`past` and `past_key_values`) be in the parameter list?

And what about the other places where I changed it, like line 570, should that be reverted too?

Reply:
Oh yeah, my comment applies for all instances of `past` that got replaced with `past_key_values`. Does the problem persist after replacing them all?

Reply:
Yes, the problem arose when I first replaced all the input_processing related code.
It will throw a ValueError, that arises from the `input_processing` method (line 436 in modeling_tf_utils.py), because `past_key_values` will remain in `kwargs_call`. That is due to the fact that `past` is not in `kwargs` passed to the `input_processing` function.  (I did some test debugging there) 
From what i could reconstruct `past_key_values` got in `kwargs_call` in the `run_call_with_unpacked_inputs` method, because it is not initially in the signature of the functions, so my first idea at a fix was to put it in there. 
Then I went a bit overboard and replaced all the `past` variables ;)

Reply:
Okay, I see what's going on! It's actually related to a [recent refactor](https://github.com/huggingface/transformers/pull/15944) we are doing on the TF side -- bringing our TF `generate()` up to speed with FLAX/PyTorch. In that PR, in the `prepare_inputs_for_generation()` function, the output dictionary key was updated from `past` to `past_key_values`. The output of that function is then fed to the model, explaining the issue you see. This is a great example of the problem of changing interfaces.

The new planned interface for `generate()` does rely on `past_key_values`, not on `past`, although most models don't use it as an explicit keyword argument. Normally, some sort of deprecation warning should be added, but since this argument is mostly for internal use (through the public `generate()`), there should be no need. I will take responsibility if a few users complain :) 

Thank you so much for having the patience to explain the rationale behind your change, it helped me understand the issue faster 💪 

(Although now I'm curious -- how did the model not break before? 🤔 )

Reply:
Yeah, I also wondered why the tests didn't break before 🤷‍♂️

Reply:
I've added a todo for me to ensure we add a new test :)